Assets/Scripts/NetworkClient.cs

```diff
@@ -249,9 +279,18 @@ async void SendClientState()
         if (isConnected())
         {
             UpdateClientState();
-            string jsonStr = JsonUtility.ToJson(_clientState);
+
+            _clientState.connection_params_dict = _connectionParams;
+
+            string jsonStr = JsonConvert.SerializeObject(_clientState, Formatting.None, _jsonSettings);
 
             _xrInputHelper.OnEndFrame();
+
+            // Stop sending connection parameters after the first message.
+            if (_connectionParams?.Count > 0)
```

Comment:
This is a little confusing. Is it possible that after parsing, _connectionParams is an empty dict? And so this Count > 0 condition doesn't yield true. And then JsonConvert.SerializeObject ends up serializing an empty (but not null) connectionParams on every call to SendClientState?

Ideally we should only include connection_params_dict in the first client state sent to the server. We shouldn't include an empty connection_params_dict in later client states.

On the server side, I will update the logic to ignore empty connection_params dicts (instead of potentially overwriting an previously-received non-empty dict).